src/s3.rs

```diff
@@ -54,6 +59,51 @@ pub fn get_client(shared_config: &aws_config::Config) -> Result<Client> {
     Ok(Client::from_conf(builder.build()))
 }
 
+#[derive(Debug)]
+pub enum ListObjectsError {
+    ListError(SdkError<ListObjectsV2Error>),
+    MissingContents(ListObjectsV2Output),
+}
+impl fmt::Display for ListObjectsError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::ListError(e) => write!(f, "Error while listing objects: {}", e),
+            Self::MissingContents(e) => {
+                write!(f, "List objects output missing contents: {:?}", e)
```

Comment:
My initial gut reaction here is that we don't want to be in the business of enumerating all the different ways in which the AWS SDK might return malformed data. I wonder if we can collapse this and future such errors into a single "Unexpected missing data in API response" error variant.

Reply:
Yeah, great question. There's the API trade off here of swallowing vs surfacing the errors that crop up during the operation. In this case I opted for surfacing, with the caller of the API able to access the original objects if that's what they want. It's definitely something that could get a bit out of hand though, and it would be nice to have patterns that could be applied across a range of use cases.

I think I'll look into `thiserror` for this, and have a bit more of a think about what we're going for before locking in this error API.